xnmt/rl/policy_gradient.py

```diff
@@ -0,0 +1,181 @@
+
+import numpy as np
+import dynet as dy
+
+from enum import Enum
+from xnmt.events import handle_xnmt_event, register_xnmt_handler
+from xnmt.transform import Linear
+from xnmt.persistence import Ref, bare, Path, Serializable, serializable_init
+from xnmt.rl.eps_greedy import EpsilonGreedy
+from xnmt.constants import EPSILON
+from xnmt.loss import FactoredLossExpr
+from xnmt.param_init import GlorotInitializer, ZeroInitializer
+
+class PolicyGradient(Serializable):
```

Comment:
Policy gradient is a training strategy, right, similar to MLE or minimum risk? I'm not sure why this class is separate and not inheriting from the parent class. What is the relationship?

Reply:
I think it's a bit different because the previous trainig strategy is a supervised one (calc_loss receive input and reference). This one is unsupervised. Maybe we should later refactor this.

Reply:
OK. I think it's OK to keep this as-is for now. My main concern is if this is floating out in a different place than the other training code then it might be an extra thing to maintain that will get stale. In a future commit let's think about a way to make sure it's tightly integrated with the trainers so we don't have duplicate code.

Reply:
Yea, I think once we have the better interface for unsupervised training, I can think of a way to integrate this.